handler/proxy.go

```diff
@@ -385,16 +412,35 @@ func (p *Proxy) SetRoundtripContext(req *http.Request, beresp *http.Response) {
 		}
 	}
 
-	for _, ctx := range attrCtx {
-		for _, ctxBody := range p.options.Context {
+	for _, ctxBody := range p.options.Context {
+		for _, ctx := range attrCtx {
+			// headers
 			options, err := NewCtxOptions(ctx, evalCtx, ctxBody)
 			if err != nil {
 				p.log.WithField("parse config", p.String()).Error(err)
 			}
 			setHeaderFields(headerCtx, options)
 		}
+
+		// query params
+		content, _, _ := ctxBody.PartialContent(schema)
+		if del, ok := content.Attributes["remove_query_params"]; ok {
+			originValue, diags := del.Expr.Value(evalCtx)
+			if diags != nil && diags.HasErrors() {
+				panic(diags.Error())
+			}
+			delQuery = append(delQuery, seetie.ValueToStringSlice(originValue)...)
+		}
+		if add, ok := content.Attributes["add_query_params"]; ok {
+			collectQueryParams(evalCtx, add, addQuery)
+		}
+		if set, ok := content.Attributes["set_query_params"]; ok {
+			collectQueryParams(evalCtx, set, setQuery)
+		}
 	}
 
+	modifyQuery(req.URL, delQuery, setQuery, addQuery)
```

Comment:
`req` could be nil here and modify query should not be called for `beresp` context calls.

Reply:
`SetRoundtripContext()` which calls the `modifyQuery()` is never called with `req = nil`.

Reply:
Thats not the point here. This method is called twice. So while reading the beresp we will apply the query params to the outreq again. I have fixed this.